src/domain/ledger/activity-checker.ts

```diff
@@ -1,49 +1,57 @@
-import { toSats } from "@domain/bitcoin"
-import { toCents } from "@domain/fiat"
-import { WalletCurrency } from "@domain/shared"
+import { AmountCalculator, WalletCurrency, ZERO_CENTS } from "@domain/shared"
 
 const MS_PER_HOUR = (60 * 60 * 1000) as MilliSeconds
 const MS_PER_DAY = (24 * MS_PER_HOUR) as MilliSeconds
 const MS_PER_30_DAYS = (30 * MS_PER_DAY) as MilliSeconds
 
+const calc = AmountCalculator()
+
 export const ActivityChecker = ({
-  dCConverter,
-  getVolumeFn,
+  getVolumeAmountFn,
   monthlyVolumeThreshold,
+  priceRatio,
 }: ActivityCheckerConfig): ActivityChecker => {
   const aboveThreshold = async (wallets: Wallet[]) => {
     const timestamp30DaysAgo = new Date(Date.now() - MS_PER_30_DAYS)
 
-    const volumeCum: TxCentsVolume = {
-      outgoingCents: toCents(0),
-      incomingCents: toCents(0),
+    const volumeCum = {
+      outgoingUsdAmount: ZERO_CENTS,
+      incomingUsdAmount: ZERO_CENTS,
     }
 
-    let incomingCents: UsdCents
-    let outgoingCents: UsdCents
-
     for (const wallet of wallets) {
-      const volume = await getVolumeFn({
+      const walletVolume = await getVolumeAmountFn({
         timestamp: timestamp30DaysAgo,
-        walletId: wallet.id,
+        walletDescriptor: wallet,
       })
-      if (volume instanceof Error) return volume
-
-      if (wallet.currency === WalletCurrency.Btc) {
-        incomingCents = dCConverter.fromSatsToCents(toSats(volume.incomingBaseAmount))
-        outgoingCents = dCConverter.fromSatsToCents(toSats(volume.outgoingBaseAmount))
-      } else {
-        incomingCents = toCents(volume.incomingBaseAmount)
-        outgoingCents = toCents(volume.outgoingBaseAmount)
-      }
-
-      volumeCum.incomingCents = toCents(volumeCum.incomingCents + incomingCents)
-      volumeCum.outgoingCents = toCents(volumeCum.outgoingCents + outgoingCents)
+      if (walletVolume instanceof Error) return walletVolume
+
+      const outgoingUsdAmount =
+        walletVolume.outgoingBaseAmount.currency === WalletCurrency.Btc
+          ? await priceRatio.convertFromBtc(
+              walletVolume.outgoingBaseAmount as BtcPaymentAmount,
+            )
+          : (walletVolume.outgoingBaseAmount as UsdPaymentAmount)
+      const incomingUsdAmount =
+        walletVolume.incomingBaseAmount.currency === WalletCurrency.Btc
+          ? await priceRatio.convertFromBtc(
+              walletVolume.incomingBaseAmount as BtcPaymentAmount,
+            )
+          : (walletVolume.incomingBaseAmount as UsdPaymentAmount)
```

Comment:
`await` is not necessary also the use of ternary op here make it difficult to read... an alternative could be (although is not necessary the change):
```typescript
      let outgoingUsdAmount = walletVolume.outgoingBaseAmount as UsdPaymentAmount
      let incomingUsdAmount = walletVolume.incomingBaseAmount as UsdPaymentAmount
      if (wallet.currency === WalletCurrency.Btc) {
        outgoingUsdAmount = priceRatio.convertFromBtc(walletVolume.outgoingBaseAmount as BtcPaymentAmount)
        incomingUsdAmount = priceRatio.convertFromBtc(walletVolume.incomingBaseAmount as BtcPaymentAmount)
      }
```

Reply:
Done https://github.com/GaloyMoney/galoy/pull/2270/commits/4e80d3b590e983e167d67c82cd9e06f8f1c7b15e